### PR TITLE
Create usersettings feature

### PR DESCRIPTION
**Pull request by roschaefer**
This is a work in progress PR for https://github.com/Human-Connection/WebApp/issues/167

### CODE REVIEW

File: features/env/services.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/* eslint no-console: off */
+const { AfterAll, BeforeAll } = require('cucumber');
+const { spawn } = require('child_process');
+
+let hcApi;
+
+// Asynchronous Callback
+BeforeAll((callback) => {
```
Comment (roschaefer) [inline features/env/services.js]: If you want to debug, comment this out

File: package.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
   },
   "scripts": {
     "clear": "rm -Rf tmp",
-    "test": "npm run eslint && cross-env NODE_ENV=test nyc --reporter=lcovonly --reporter=text-summary npm run mocha",
+    "test": "npm run eslint && cross-env NODE_ENV=test nyc --reporter=lcovonly --reporter=text-summary npm run mocha && npm run cucumber",
     "test:mongo": "npm run eslint && cross-env NODE_ENV=test nyc --reporter=lcovonly --reporter=text-summary npm run mocha:mongo",
     "eslint": "eslint server/. test/. --config .eslintrc.json",
     "start": "concurrently 'mongod' 'wait-on tcp:27017 && cross-env NODE_ENV=production node server/'",
```
Comment (roschaefer) [inline package.json]: In the script `dev:debug` the `NODE_ENV=development` is hard coded. If you want to run the backend separately from cucumber tests (for debugging) make sure you change `NODE_ENV=test` so both the cucumber test and your backend running in the background are using the `test` database.

**Comment by roschaefer** [comment]
My current progress on debugging the error "There is no current user to associate" is that on [this line here](https://github.com/Human-Connection/API/blob/develop/server/services/usersettings/usersettings.hooks.js#L11), the supplied `context` is different.

In development environment:
```
context.params
{query: {…}, provider: "socketio", payload: {…}, user: {…}, accessToken: "eyJhbGciOiJIUzI1NiIsInR5cCI6ImFjY2VzcyIsInR5cGUiOi…dmIn0.jeBwCvQCGG_dRRgvoU3jTSc10bnZyZ2X89I2KP9D6HQ", …}accessToken: "eyJhbGciOiJIUzI1NiIsInR5cCI6ImFjY2VzcyIsInR5cGUiOiJhY2Nlc3MifQ.eyJ1c2VySWQiOiI1YjMxM2EzZjdkZGY5NjcyMGU0NTcxZTciLCJpYXQiOjE1MzAzMTg1OTgsImV4cCI6MTUzMDQwNDk5OCwiYXVkIjoiaHR0cDovL2xvY2FsaG9zdCIsImlzcyI6ImZlYXRoZXJzIiwic3ViIjoiYW5vbnltb3VzIiwianRpIjoiM2ZlNTYxMmQtOWY4Mi00NTlmLWJiM2QtYThmOGNkYjFhYzdmIn0.jeBwCvQCGG_dRRgvoU3jTSc10bnZyZ2X89I2KP9D6HQ"authenticated: trueheaders: {Authorization: "eyJhbGciOiJIUzI1NiIsInR5cCI6ImFjY2VzcyIsInR5cGUiOi…dmIn0.jeBwCvQCGG_dRRgvoU3jTSc10bnZyZ2X89I2KP9D6HQ"}mongoose: {upsert: true}payload: {userId: "5b313a3f7ddf96720e4571e7"}provider: "socketio"query: {}user: {_id: ObjectID, email: "test@test.de", password: "$2a$12$7b8GrkayDpdtP9t548eLdO7Wi.2Dib9IfjRzoiLXWGZa8MwrzKely", name: "Peter", slug: "peter", …}__proto__: Object
```

In testing environment:
```
context.params
{query: {…}, provider: "rest", headers: {…}, mongoose: {…}}headers: {content-type: "application/json", authorization: "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6ImFjY2VzcyIsInR…JkIn0.YU7bVzxS9mmjdkZVfkM5Pk845w3vqoDX_kuyeCOOQek", accept: "*/*", content-length: "55", user-agent: "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)", …}mongoose: {upsert: true}provider: "rest"query: {}__proto__: Object
```

Note that in development environment `context.params.user` is given, while in testing environment `context.params.user` is `undefined`.

@appinteractive could it be that sending ordinary HTTP requests with node fetch result in the undefined `context.params.user` here?

**Comment by appinteractive** [comment]
It does not look for params.userId but for data.userId. Maybe that’s the issue?

**Comment by appinteractive** [comment]
Take a look at the authentication tests which uses the parameters to specify which protocol to use:
https://github.com/Human-Connection/API/blob/develop/test/services/authentication.js

**Pull request merged**